packages/dds/tree/src/feature-libraries/editableTreeBinder.ts

```diff
@@ -3,143 +3,12 @@
-}
-
-/**
- * Binder events signaling state invalidation
- */
-export interface InvalidationBinderEvents {
-	invalidation(context: InvalidationBindingContext): void;
-}
-
-/**
- * Compare function, generic.
- */
-export type CompareFunction<T> = (a: T, b: T) => number;
-
-/**
- * Compare function for binder events.
- */
-export type BinderEventsCompare = CompareFunction<VisitorBindingContext>;
-
-/**
- * Compare function for anchors.
- */
-export type AnchorsCompare = CompareFunction<UpPath>;
-
-/**
- * Options to configure binder behavior.
- *
- * TODO: add more options:
- * `filterFn?: (context: BindingContext) => boolean;`
- * `pathPolicy?: "relative" | "absolute";`
- */
-export interface BinderOptions {
-	sortFn?: BinderEventsCompare;
-}
-
-/**
- * Specialized binder options for flushable binders.
- */
-export interface FlushableBinderOptions<E extends Listeners<E>> extends BinderOptions {
-	autoFlush: boolean;
-	autoFlushPolicy: keyof Listeners<E>;
-	sortAnchorsFn?: AnchorsCompare;
-}
-
-/**
- * Match policy for binding: subtree or path.
- *
- * - `subtree` match policy means that path filtering would return events matching the exact path and its subpaths,
- * ie. changes to (nested) children would be allowed to bubble up to parent listeners.
- * - {@link SubtreePolicy} match policy is  equivalent with `subtree` match policy, while allowing to specify a maximum
- * depth for the subtree.
- * - `path` match policy means that path filtering would return events matching the _exact_ path only. In this case
- * _exact_ semantics include interpreting an `undefined` _index_ field in the {@link PathStep} as a wildcard.
- *
- */
-export type MatchPolicy = SubtreePolicy | "subtree" | "path";
-
-/**
- * Subtree match policy where max depth can be specified.
- */
-export interface SubtreePolicy {
-	maxDepth: number;
-}
-
-/**
- * The data binder interface
- */
-export interface DataBinder<B extends OperationBinderEvents | InvalidationBinderEvents> {
-	/**
-	 * Register an event listener
-	 *
-	 * @param anchor - The anchor to register the listener on
-	 * @param eventType - The {@link BindingType} to listen for.
-	 * @param eventTrees - The {@link BindPolicy}s to filter on.
-	 * @param listener - The listener to register
-	 */
-	register<K extends keyof Listeners<B>>(
-		anchor: FlexTreeNode,
-		eventType: K,
-		eventTrees: BindPolicy[],
-		listener?: B[K],
-	): void;
-
-	/**
-	 * Unregister all listeners.
-	 */
-	unregisterAll(): void;
-}
-
-/**
- * An interface describing the ability to flush.
- */
-export interface Flushable<T> {
-	flush(): T;
-}
-
-/**
- * An interface describing a flushable data binder.
- */
-export interface FlushableDataBinder<
-	B extends OperationBinderEvents | InvalidationBinderEvents,
-> extends DataBinder<B>,
-		Flushable<FlushableDataBinder<B>> {}
+import { type FieldKey, type UpPath, topDownPath } from "../core/index.js";
 
 /**
  * A step in a bind path
  */
-export interface PathStep {
+interface PathStep {
```

Comment:
DownPath, which is exported, references PathStep

Reply:
I considered doing that. We don't require types referenced as implementation details to be exported except at package boundaries. Given this is only used in one place which is a test and we have no plans to use it more, I don't think we really need to export is here (or in the index file)

Reply:
When trying to trim down dead code, exporting more things that we need makes unused things (like this entire file other than this) hard to find, so I'm erroring on the side of exporting less. Its easier to adding missing exports than remove unneeded ones, so they tend to pile up.